lightwood/mixers/nn/nn.py

```diff
@@ -67,29 +70,42 @@ def predict(self, when_data_source, include_encoded_predictions = False):
                 if CONFIG.SELFAWARE:
                     output, awareness = self.net(inputs)
                     awareness = awareness.to('cpu')
+                    awareness_arr.extend(awareness)
                 else:
                     output = self.net(inputs)
+                    awareness_arr = None
+
                 output = output.to('cpu')
 
             outputs.extend(output)
 
         output_trasnformed_vectors = {}
-
-        for output_vector in outputs:
+        confidence_trasnformed_vectors = {}
+
+        for i in range(len(outputs)):
+            if awareness_arr is not None:
+                confidence_vector = awareness_arr[i]
+                transformed_confidence_vectors = when_data_source.transformer.revert(confidence_vector,feature_set = 'output_features')
```

Comment:
not sure, what the logic is here, but the confidence vector isn't the actual value but rather the distance to the target, lets discuss intervals